core/dbt/task/init.py

```diff
@@ -274,9 +276,18 @@ def run(self):
             in_project = False
 
         if in_project:
+            # If --profile was specified, it means use an existing profile, which is not
+            # applicable to this case
+            if getattr(get_flags(), "PROFILE", None):
```

Comment:
`self.args` will give you access to the profile flag's value, so use that instead of `getattr(get_flags(), "PROFILE", None)`, as this is how we do it elsewhere in core tasks